compiler/rustc_const_eval/src/interpret/call.rs

```diff
@@ -614,7 +614,7 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
             // codegen'd / interpreted as virtual calls through the vtable.
             ty::InstanceKind::Virtual(def_id, idx) => {
                 let mut args = args.to_vec();
-                // We have to implement all "dyn-compatible receivers". So we have to go search for a
+                // We have to implement all "dyn compatible receivers". So we have to go search for a
```

Comment:
Doesn't English grammar say we need a `-` since this is a compound term used as an adjective?

Reply:
Travis wrote there:
> one would probably say that "a trait is dyn compatible" (but one might "have a dyn-compatible trait")

This is what makes most sense to me. A compound term in English is used with hyphen in adjective form but without hyphen elsewhere. That's how English works, it would be non-grammatical and confusing to insist on the same spelling everywhere.

Reply:
Huh, I don't know what the "formal" English answer to this is (as best there is one with a descriptive language).

[This SFBar site](https://www.sfbar.org/blog/to-hyphenate-or-not-to-hyphenate/#:~:text=Some%20compound%20terms%20are%20written,such%20as%20fireworks%20and%20upstream) seems to suggest that it's a per-term decision.  I personally understand "dyn-compatible" to be one particular compound hyphenated term which I would not expect to see split up. 

[APA style](https://apastyle.apa.org/learn/faqs/when-use-hyphen) says "If a compound adjective can be misread, use a hyphen." As an example:

> "the adolescents resided in two parent homes" means that two homes served as residences, whereas if the adolescents resided in "two-parent homes," they each would live in a household headed by two parents.

However, it also says:

> Most compound adjective rules are applicable only when the compound adjective precedes the term it modifies. If a compound adjective follows the term, do not use a hyphen, because relationships are sufficiently clear without one.

e.g. "client-centered counseling" but "the counseling was client centered"

Which seems to suggest that we could avoid the hyphen when the compound adjective follows the term ("the trait is dyn compatible").

It doesn't seem to me like there's an obvious answer here. Personally, I'd prefer to apply the general principle of using a hyphen to avoid confusion when referring to a novel compound term like this.